src/lib/comms/sol-oic-client.c

```diff
@@ -910,7 +910,7 @@ _resource_request(struct sol_oic_client *client, struct sol_oic_resource *res,
 
 SOL_API bool
 sol_oic_client_resource_request(struct sol_oic_client *client, struct sol_oic_resource *res,
```

Comment:
Instead of adding another parameter to this function, I'd make another public function only for non-confirmable requests. This function already has way too many parameters.